### ISSUE TITLE
Latest Alpine Docker Image fails to build.

### ISSUE THREAD

**Issue by davidzech** [simulationcraft/simc#7078]
### Describe the bug
`alpine:latest` fails to install the dependencies listed in `Dockerfile`. As of writing `latest` resolves to `3.17.x`. 
`build_dependencies` and `compiler-rt-static` fail to install.

### Expected behavior
Docker builds should succeed.

### To Reproduce
```sh
docker build https://github.com/simulationcraft/simc.git\#dragonflight   
```

outputs:
```sh
→ docker build https://github.com/simulationcraft/simc.git\#dragonflight                       
[+] Building 2.1s (6/14)                                                                                                                                                                                                                                                                            
 => CACHED [internal] load git source https://github.com/simulationcraft/simc.git#dragonflight                                                                                                                                                                                                 0.0s
 => [internal] load metadata for docker.io/library/alpine:latest                                                                                                                                                                                                                               1.3s
 => [build 1/8] FROM docker.io/library/alpine:latest@sha256:8914eb54f968791faf6a8638949e480fef81e697984fba772b3976835194c6d4                                                                                                                                                                   0.0s
 => => resolve docker.io/library/alpine:latest@sha256:8914eb54f968791faf6a8638949e480fef81e697984fba772b3976835194c6d4                                                                                                                                                                         0.0s
 => CACHED [build 2/8] COPY . /app/SimulationCraft/                                                                                                                                                                                                                                            0.0s
 => CACHED [stage-1 2/5] RUN apk --no-cache add --virtual build_dependencies         libcurl         libgcc         libstdc++                                                                                                                                                                  0.0s
 => ERROR [build 3/8] RUN apk --no-cache add --virtual build_dependencies             compiler-rt-static             curl-dev             clang-dev             llvm             g++             make             git                                                                          0.8s
------                                                                                                                                                                                                                                                                                              
 > [build 3/8] RUN apk --no-cache add --virtual build_dependencies             compiler-rt-static             curl-dev             clang-dev             llvm             g++             make             git:                                                                                     
#6 0.162 fetch https://dl-cdn.alpinelinux.org/alpine/v3.17/main/aarch64/APKINDEX.tar.gz
#6 0.455 fetch https://dl-cdn.alpinelinux.org/alpine/v3.17/community/aarch64/APKINDEX.tar.gz
#6 0.772 ERROR: unable to select packages:
#6 0.794   build_dependencies-20221130.005922:
#6 0.794     masked in: cache
#6 0.794     satisfies: world[build_dependencies=20221130.005922]
#6 0.794   compiler-rt-static (no such package):
#6 0.794     required by: build_dependencies-20221130.005922[compiler-rt-static]
------
process "/bin/sh -c apk --no-cache add --virtual build_dependencies             compiler-rt-static             curl-dev             clang-dev             llvm             g++             make             git" did not complete successfully: exit code: 2
```

Forcing `linux/amd64` also has the same:
```sh
→ docker build --platform=linux/amd64 https://github.com/simulationcraft/simc.git\#dragonflight
[+] Building 2.7s (6/14)                                                                                                                                                                                                                                                                            
 => CACHED [internal] load git source https://github.com/simulationcraft/simc.git#dragonflight                                                                                                                                                                                                 0.0s
 => [internal] load metadata for docker.io/library/alpine:latest                                                                                                                                                                                                                               0.8s
 => CACHED [build 1/8] FROM docker.io/library/alpine:latest@sha256:8914eb54f968791faf6a8638949e480fef81e697984fba772b3976835194c6d4                                                                                                                                                            0.0s
 => => resolve docker.io/library/alpine:latest@sha256:8914eb54f968791faf6a8638949e480fef81e697984fba772b3976835194c6d4                                                                                                                                                                         0.0s
 => CACHED [build 2/8] COPY . /app/SimulationCraft/                                                                                                                                                                                                                                            0.0s
 => ERROR [build 3/8] RUN apk --no-cache add --virtual build_dependencies             compiler-rt-static             curl-dev             clang-dev             llvm             g++             make             git                                                                          1.9s
 => CANCELED [stage-1 2/5] RUN apk --no-cache add --virtual build_dependencies         libcurl         libgcc         libstdc++                                                                                                                                                                1.9s
------                                                                                                                                                                                                                                                                                              
 > [build 3/8] RUN apk --no-cache add --virtual build_dependencies             compiler-rt-static             curl-dev             clang-dev             llvm             g++             make             git:                                                                                     
#6 0.272 fetch https://dl-cdn.alpinelinux.org/alpine/v3.17/main/x86_64/APKINDEX.tar.gz                                                                                                                                                                                                              
#6 0.986 fetch https://dl-cdn.alpinelinux.org/alpine/v3.17/community/x86_64/APKINDEX.tar.gz                                                                                                                                                                                                         
#6 1.761 ERROR: unable to select packages:
#6 1.832   build_dependencies-20221130.010208:
#6 1.832     masked in: cache
#6 1.832     satisfies: world[build_dependencies=20221130.010208]
#6 1.832   compiler-rt-static (no such package):
#6 1.832     required by: build_dependencies-20221130.010208[compiler-rt-static]
------
process "/bin/sh -c apk --no-cache add --virtual build_dependencies             compiler-rt-static             curl-dev             clang-dev             llvm             g++             make             git" did not complete successfully: exit code: 2
```

**Comment by davidzech** [comment]
Pinning the base image to `3.16.3` should  be sufficient.

**Comment by cesoun** [comment]
Out of curiosity (and not knowing a whole lot about simc under the hood or `compiler-rt-static`), is there a reason this couldn't use `compiler-rt` instead of `compiler-rt-static`?

Alpine Linux has `compiler-rt` released up to edge at the moment with latest (3.17) [found here](https://pkgs.alpinelinux.org/packages?name=compiler-rt*&branch=v3.17&repo=&arch=&maintainer=). The `compiler-rt-static` release is only present (as of right now) up to Alpine 3.15 [found here](https://pkgs.alpinelinux.org/packages?name=compiler-rt*&branch=v3.15&repo=&arch=&maintainer=).

Updating `compiler-rt-static` in the Dockerfile to `compiler-rt` seems to work just fine.

https://github.com/simulationcraft/simc/blob/c7727352821e5ff9a6c5c3be3b823ac4d859f578/Dockerfile#L35-L42

**Comment by Bloodmallet** [comment]
Greetings,
thank you for reporting this issue. I didn't have access to docker in a long time but will regain access some time next week. I'll have a look at this issue once I have.

About compiler settings/flags/features: I don't know anything about them. I got the current settings via direct messages from people more well versed in this matter and tried them out until I found a working setup that seemed to perform well. I'll happily return to fix the image. If you have suggestions on how to fix it or how to improve the way the image is created/compiled, please create a PR. I'm more than happy to check out your suggestions!

**Comment by Bloodmallet** [comment]
I'm looking into this right now. As mentioned above I'm not knowledgeable about compilers and their features. @scamille what's your opinion about the above change from using `compiler-rt-static` to `compiler-rt`? I'm trying this locally and it's fixing the broken image.

On a different note, I'm slightly changing the Dockerfile. While I wasn't actively working with this tech the past few years I picked up minor things left and right and am adding these.

1. Merging layers
2. Echoing stages
3. Changing how Variables are called. `{$MY_VARIABLE}` to `${MY_VARIABLE}`
4. Adding an ENTYPOINT and  CMD (Entrypoint provides the simc executable, cmd is just a placeholder .simc file)

Because I can't break it more than it currently is, I'm pushing my changes. Everyone, please check them out.

EDIT: I'm aware that adding `./simc` as an entrypoint will probably break peoples pipelines. But I hope this change improves the usage of the image. If you're just using the image and building on top of it, this doesn't affect you.

Please describe your use-cases if you're NOT primarily interested in using the simc executable. To hopefully ease the transition here are two examples about how to change a call after this change:
Original:
```
docker run simulationcraftorg/simc:latest ./simc my_simc_file.simc
```
New:
```
docker run simulationcraftorg/simc:latest my_simc_file.simc
```

Original:
```
docker run simulationcraftorg/simc:latest non_simc_command
```
New:
```
docker run simulationcraftorg/simc:latest --entrypoint non_simc_command
```

**Comment by scamille** [comment]
I noticed you triggered https://github.com/simulationcraft/simc-publish/actions/runs/3668532005/jobs/6201878989 - and it looks like it works again now :-)

**Comment by davidzech** [comment]
Fixed in 7461bec0dbbfcdb826a655ca7ce053857834f75c
